test/state/state.cpp

```diff
@@ -181,6 +182,49 @@ std::variant<TransactionReceipt, std::error_code> transition(
         return acc.destructed || (rev >= EVMC_SPURIOUS_DRAGON && acc.erasable && acc.is_empty());
     });
 
-    return TransactionReceipt{gas_used, host.take_logs()};
+    auto receipt = TransactionReceipt{tx.kind, result.status_code, gas_used, host.take_logs(), {}};
```

Comment:
@chfast Reviewing it for the last time I changed initialisation of receipt status filed to `result.status_code` from `EVMC_SUCCESS`. I think the prev version was wrong.